homeassistant/components/google_assistant/smart_home.py

```diff
@@ -255,11 +261,15 @@ def async_update(self):
 
         devices.append(serialized)
 
-    return {
+    response = {
         'agentUserId': config.agent_user_id,
         'devices': devices,
     }
 
+    hass.bus.async_fire(EVENT_SYNC_RECEIVED, response)
```

Comment:
It should be one event, and it should be fired before doing an action or returning data to Google. That way we can tie it all together in the future using context. 

This is the same as how we do automations, alexa and script.

The reason other integrations don't have this event, is that other integrations don't read our data. It's good for the user to be able to understand what data of them flows to the cloud.

Reply:
Fixed, only one sync event now sent and fired before action.